OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsActivity.java

```diff
@@ -95,7 +101,40 @@ public boolean onPreferenceClick(Preference preference) {
             initializeUseNumKeypadForYubiKeyPin(
                     (CheckBoxPreference) findPreference(Constants.Pref.USE_NUMKEYPAD_FOR_YUBIKEY_PIN));
 
+        } else if (action != null && action.equals(ACTION_PREFS_GUI)) {
+            addPreferencesFromResource(R.xml.gui_preferences);
+
+            initializeTheme((ListPreference) findPreference(Constants.Pref.THEME));
+        }
+    }
+
+    @Override
```

Comment:
The following code is repeated in BaseActivity. Could you externalize this into it's own class?